Tribler/Core/TFTP/handler.py

```diff
@@ -454,6 +476,12 @@ def _handle_packet_as_receiver(self, session, packet):
         self._logger.debug(u"%s Got data, #block = %s size = %s", session, packet['block_number'], len(packet['data']))
 
         # check block_number
+        # ignore old ones, they may be retransmissions
+        if packet['block_number'] < session.block_number:
```

Comment:
What's happening here? Are you enforcing the packets such that they arrive in order? You know that's not how the internet works right?